bitnami/kafka/templates/svc.yaml

```diff
@@ -18,25 +18,11 @@ spec:
   {{- end }}
   {{- end }}
   ports:
-    - name: tcp-kafka
+    - name: kafka
       port: {{ .Values.service.port }}
-      protocol: TCP
       targetPort: kafka
-      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.kafka)) }}
-      nodePort: {{ .Values.service.nodePorts.kafka }}
-      {{- else if eq .Values.service.type "ClusterIP" }}
-      nodePort: null
+      {{- if eq .Values.service.type "NodePort" }}
+      nodePort: {{ .Values.service.nodePort }}
```

Comment:
When you use a "LoadBalancer" service, a "nodePort" is also created (and associated to the external IP obtained via load balancer). Therefore, this should be:
```
    {{- if or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer") }}
```
Then, you're assuming that users will set the `.Values.service.nodePort` and that won't be true on many cases, so you have two alternatives:
- Generate a random `nodePort` when it's not specified (which implies reverting your changes)
- Make it mandatory, which I don't think it's necessary:

```
    nodePort: {{ required "A nodePort is mandatory!" .Values.service.nodePort }}
```

Reply:
Please do not remove this block:
```
      {{- else if eq .Values.service.type "ClusterIP" }}
      nodePort: null
```

Reply:
I didn't know about LoadBalancers getting nodePorts. Two questions:

- Why aren't we following the same approach for svc-external-access.yaml by allowing them to pass nodePorts as well as loadBalancerIPs? Or the opposite question why don't we set nodePort to null in svc.yaml (like we do in svc-external-access.yaml) when it is of type LoadBalancer?
- Why is the `nodePort: null` required for type `ClusterIP` isn't that the default?

Reply:
The `nodePort: null` is related to some weird behaviour reported by some users.
Regarding the "external-service", you need to chose whether configuring Kafka advertised listeners based on the nodeports or based on the external IPs (associated to LoadBalancer svc), that's why we only allow setting one of them.